src/key_reset.c

```diff
@@ -925,9 +931,15 @@ PEP_STATUS create_standalone_key_reset_message(PEP_SESSION session,
 
     message* output_msg = NULL;
     
-    status = encrypt_message(session, reset_msg, NULL,
-                             &output_msg, PEP_enc_auto,
-                             PEP_encrypt_flag_key_reset_only | PEP_encrypt_flag_force_no_attached_key);
+    PEP_encrypt_flags_t flags =
+       PEP_encrypt_flag_key_reset_repropagate
+       | PEP_encrypt_flag_key_reset_only
+       | PEP_encrypt_flag_force_no_attached_key;
+
+    if (!reset_msg->from->fpr) {
+        reset_msg->from->fpr = old_fpr;
```

Comment:
`old_fpr` must be `strdup`ed, otherwise you risk double-frees.

Reply:
Ooops, that might be where the double free comes from. Good catch! Thanks!